internal/hcs/system.go

```diff
@@ -326,13 +336,113 @@ func (computeSystem *System) Properties(ctx context.Context, types ...schema1.Pr
 	return properties, nil
 }
 
+// statisticsInProc emulates what HCS does to grab statistics for a given container with a small
+// change to make grabbing the private working set total much more efficient.
+func (computeSystem *System) statisticsInProc(ctx context.Context) (*hcsschema.Properties, error) {
+	// See if we'd even be able to open the silo. We'll need to be local
+	// system/system so check first.
+	usr, err := windows.GetCurrentProcessToken().GetTokenUser()
+	if err != nil {
+		return nil, err
+	}
+	if !usr.User.Sid.IsWellKnown(windows.WinLocalSystemSid) {
+		return nil, errors.New("process does not have the right permissions to open the silo")
+	}
+
+	jobOptions := &jobobject.Options{
+		UseNTVariant: true,
+		Name:         siloNameFmt(computeSystem.id),
```

Comment:
WS2022+ supports passing a job object to HCS for it to use for the silo. Maybe in the future we can take advantage of that.

Reply:
Especially since this current approach is brittle if the name format changes.

Reply:
Oh sweet! I asked Kris what builds this was supported in but was waiting to hear back

Reply:
Actually looks like I might be wrong about WS2022 having support.

Reply:
> WS2022+ supports passing a job object to HCS for it to use for the silo. Maybe in the future we can take advantage of that.

WHAT! You mean I can pick the name or I can pick the parent?

Reply:
Afaik from checking, the naming conventions haven't changed. They still need to be named in this format :(

Reply:
The work allows you to pass a jo handle directly though so I suppose this could be a nested job potentially..